### DISCUSSION TITLE
Is there a way of seeing which plugins have been added?

### DISCUSSION THREAD

**Discussion by venkatab144** [react-page/react-page#1055] [Q&A]
I have a custom plugin and I want to be able to store all instances of this plugin that have been added so that I can get their values. For example, I have plugin X that has property Y, the user adds an 'n' number of plugin X and I want to be able to access all these plugin X's so that I can get their property Y. One way I thought was looking through the JSON and finding "id": "PluginX" and checking its properties but this gets a bit complex. Is there a way of tracking added plugins?

**Comment by macrozone** [comment]
I would probably also iterate over the value, its not that efficient, but you probably won't notice. 

There is some code in the editor that actually does that (e.g. migration code), but maybe it would be worth in the future to export a small helper function that walks the tree.

But its actually not complictated, iterate over all rows if its a cell, then iterate over all cells if its a row. do that recursivly. Should be easy if you use typescript as the `Value` is fully typed.
